askama_parser/src/expr.rs

```diff
@@ -927,6 +927,27 @@ impl<'a> Suffix<'a> {
             }
         }
 
+        fn lifetime<'a>(i: &mut InputStream<'a>) -> ParseResult<'a, ()> {
+            // Before the 2021 edition, awe can have whitespace characters between "r#" and the
+            // identifier so we allow it here.
```

Comment:
I would still reject `'r#<space>`. We are `--edition 2024`, so a user should not expect that their `--edition 2018` code works without changes. (At least for such esoteric cases.) In `--edition 2021` it was already wrong: <https://godbolt.org/z/8WbrPxv83>

Reply:
We are, but the codebase using askama might not be.

Reply:
Can you please add a test what happens if the template code contains `{{ x!( #r' y )}}`? Only to make sure that it does not panic.

Reply:
Sure, done!